### PR TITLE
Enabling retrieval of last reading sessions progress

### PR DESCRIPTION
**Pull request by tfnribeiro**
**Changes**
- Created endpoint `get_user_unfinished_reading_sessions` to retrieve the unfinished sessions by a user
- Added function to query the user_activity_data to retrieve candidates for unfinished sessions
- Added a function that takes a list of scroll events and returns the maximum "actual" scrolled to percentage. This is done by calculating the velocity between scrolls, and having a threshold for the maximum value that can be updated from one update to the next.

### CODE REVIEW

**Comment by tfnribeiro** [comment]
@mircealungu Now it rendered! Even though it came from my branch, but I still don't get the arrows? 

For the Unit testing, I think we need to update the setup due to the `punkt_tab` for NLTK requirement, I have made update in: https://github.com/zeeguu/api/pull/187

File: zeeguu/api/endpoints/user.py [hunk 1]
```diff
@@ -88,6 +88,36 @@ def learned_and_native_language():
     res = dict(native=user.native_language_id, learned=user.learned_language_id)
     return json_result(res)
 
+@api.route("/get_user_unfinished_reading_sessions", methods=("GET",))
```
Comment (mircealungu) [inline zeeguu/api/endpoints/user.py:91]: the concept is `user_reading_session` so the name should maybe follow that? i.e., `get_unfinished_user_reading_sessions`?

File: zeeguu/core/model/user_activitiy_data.py [hunk 2]
```diff
@@ -291,6 +292,53 @@ def get_article_id(self, db_session):
...
+    @classmethod
+    def get_scroll_events_for_user_in_date_range(cls, user, days_range=7, limit=1):
+        """
+            Returns a list of parsed scroll user events containing the last point which was "read"
+
+            [(article_id:int, string_date:str, viewPortSettings:dict, )]
+        """
+        def seconds_to_hour(sec):
+            return sec // 60 // 60
+        current_date = (datetime.now() + timedelta(1)).date()
```
Comment (mircealungu) [inline zeeguu/core/model/user_activitiy_data.py:304]: is this not `tomorrows_date`?

File: zeeguu/core/util/scroll_last_reading_percentage.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            
+        The method goes through the list of events and calculates how much % of the article the 
+        user scrolls past in a second. If the user scrolls more than 10% in a second, it's not 
+        considered that they have read that ammount of the article.
+
+        However, let's say they scrolled quickly through the top image, and it's a rather small article,
+        then if the speed is quite low, and they haven't scrolled more than 40% of the last point,
+        then we consider that to be a valid update. 
+
+        There's a grace of 5% meaning if the user as gotten to 95% of the article, then it's considered
```
Comment (mircealungu) [inline zeeguu/core/util/scroll_last_reading_percentage.py:21]: makes sense.

File: zeeguu/core/util/scroll_last_reading_percentage.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+def find_last_reading_percentage(list, max_jump=10, max_total_update=40):
```
Comment (mircealungu) [inline zeeguu/core/util/scroll_last_reading_percentage.py:1]: we could almost create a `user_behavior` or `behavioral_estimation`, or `behavioral_modeling` module or something similar to put this there? it's more central to the app than just a `util`.

**Review by mircealungu — APPROVED** [review]
overall looks good. 
just minor observations and asks :)

**Pull request merged**